src/main/java/com/ldtteam/structurize/util/InstantStructurePlacer.java

```diff
@@ -94,63 +128,81 @@ public void setupStructurePlacement(@NotNull final BlockPos pos, final boolean c
     /**
      * Place a structure into the world.
      *
-     * @param world the placing player.
-     * @param storage the change storage.
+     * @param world    the placing player.
+     * @param storage  the change storage.
      * @param inputPos the start pos.
      * @return the last pos.
      */
     public BlockPos placeStructure(final World world, final ChangeStorage storage, final BlockPos inputPos)
+    {
+        return placeStructure(world, storage, inputPos, (structure, pos) -> structure.getBlockState(pos).getBlock() == ModBlocks.blockSubstitution
+                                                                           || structure.getBlockState(pos).getBlock() instanceof IAnchorBlock);
+    }
+
+    /**
+     * 
+     * @param world             the placing player.
+     * @param storage           storage the change storage.
+     * @param inputPos          the start pos.
+     * @param skipIfNotComplete a function that determines whether to skip a block
+     *                          if complete is false.
+     * @return the last pos.
+     */
+    public BlockPos placeStructure(final World world, final ChangeStorage storage, final BlockPos inputPos, BiFunction<Structure, BlockPos, Boolean> skipIfNotComplete)
     {
         structure.setLocalPosition(inputPos);
         @NotNull final List<BlockPos> delayedBlocks = new ArrayList<>();
         final BlockPos endPos = new BlockPos(this.structure.getWidth(), this.structure.getHeight(), this.structure.getLength());
-        BlockPos currentPos = inputPos;
+        BlockPos currentPos = new BlockPos(inputPos.getX(), inputPos.equals(BlockPos.ZERO) ? endPos.getY() - 1 : inputPos.getY(), inputPos.getZ());
         int count = 0;
 
-        for (int y = currentPos.getY(); y < endPos.getY(); y++)
+        for (int y = currentPos.getY(); y >= 0; y--)
```

Comment:
You can't iterate from up to down because of falling blocks, or make them go into delayed blocks (and even that is not a complete solution)

Reply:
how does iterating from the top cause any problems with falling blocks?
it shouldn't create any the from the bottom up wouldn't have created, and both would ignore falling blocks after that.

Reply:
when you place a falling block and it has no support it falls down, and most of falling blocks are solid so they don't go into the delayed blocks

Reply:
oh, now i understand, yes thats a problem, i thought with blocks that should get broken and can fall.

Reply:
what about just adding all the fluid blocks to remove to delayedBlocks?
That all gets handled in a single tick, so the order wouldn't be a problem for fluids.

Reply:
oops, they don't get handled all in the same tick.

Reply:
you would have to prioritize them because delayedBlocks are mostly non-solid blocks, also now imagine you have lava block in the structure and it becomes obsidian.
btw this placing mechanism doesn't cover many problematic situations.
I would say the best solution which does not mess up other things is to simply clear all fluids that do not match the structure in a single tick as the first step

Reply:
yes, thats probably right, i thought all the delayed blocks would get handled in a single tick, then it might have worked, but not like this.

Reply:
but i don't think that with the obsidian would happed anyways.

Reply:
I did that, i am just not sure whether its a good idea to only remove fluids if the input pos is 0,0,0, or if that will just cause strange bugs again.

Reply:
You can actually iterate from up to down in while removing fluid, but don't do that when placing

Reply:
and rares cases are still cases :P

Reply:
> You can actually iterate from up to down in while removing fluid, but don't do that when placing

yes, but it doesn't matter.
as this is all in a single tick, none of the fluids can flow, so it can be ignored.

> and rares cases are still cases :P

yes, but i did some tests and the overall performance was better without that check, as he has to check that way more often, then he could actually keep the fluid.